File: src/1-Presentation/simplecrm.attendant.webapp/src/domain/models/api/responses/InteractionRS.ts
import {UserRS} from "./UserRS";
import {CustomerRS} from "./CustomerRS";
import {BaseRS} from "./BaseRS";

export interface InteractionRS extends BaseRS{
    id: string,
    userId: string,
    customerId: string,
    creationTime: Date,
    endTime: Date,
    user: UserRS,
    customer: CustomerRS
}

File: src/2-Application/SimpleCRM.Application.Common.Contracts/DTOs/InteractionRS.cs
﻿namespace SimpleCRM.Application.Common.Contracts.DTOs;

public class InteractionRS : BaseRS
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public Guid CustomerId { get; set; }
    public DateTime CreationTime { get; set; }
    public DateTime EndTime { get; set; }
    public UserRS User { get; set; } = default!;
    public CustomerRS Customer { get; set; } = default!;
}